packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol

```diff
@@ -202,13 +211,19 @@ contract ProtectedDataSharing is ERC721Burnable, ERC721Receiver, ManageOrders, A
     //protectedData's owner should approve this SC
     function addProtectedDataToCollection(
         uint256 _collectionId,
-        address _protectedData
+        address _protectedData,
+        address _appAddress
     ) public onlyCollectionOwner(_collectionId) {
+        require(_appAddress != address(0), "App address invalid");
+        appForProtectedData[_collectionId][_protectedData] = _appAddress;
         uint256 tokenId = uint256(uint160(_protectedData));
-        require(registry.getApproved(tokenId) == address(this), "Collection Contract not approved");
-        registry.safeTransferFrom(msg.sender, address(this), tokenId);
+        require(
+            protectedDataRegistry.getApproved(tokenId) == address(this),
+            "ProtectedDataSharing Contract not approved"
+        );
+        protectedDataRegistry.safeTransferFrom(msg.sender, address(this), tokenId);
         protectedDatas[_collectionId][uint160(_protectedData)] = _protectedData;
-        emit ProtectedDataAddedToCollection(_collectionId, _protectedData);
+        emit ProtectedDataAddedToCollection(_collectionId, _protectedData, _appAddress);
     }
 
     // TODO: Should check there is no subscription available and renting
```

Comment:
add a related ticket or comment this method (unsafe as is)

Reply:
ticket created with Charlène